beacon-chain/core/blocks/block_operations_test.go

```diff
@@ -0,0 +1,146 @@
+package blocks
+
+import (
+	"fmt"
+	"testing"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+func TestIncorrectProcessProposerSlashings(t *testing.T) {
+	// We test exceeding the proposer slashing threshold.
+	slashings := make([]*pb.ProposerSlashing, params.BeaconConfig().MaxProposerSlashings+1)
+	registry := []*pb.ValidatorRecord{}
+	currentSlot := uint64(0)
+
+	if _, err := ProcessProposerSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); err == nil {
+		want := "number of proposer slashings exceeds threshold"
+		t.Errorf("Expected %v, received nil", want)
+	}
+	currentSlot++
+
+	// We now test the case with unmatched slot numbers.
```

Comment:
I think each of these test scenarios should be its own test 

```
func TestProcessProposerSlashings_UnmatchedSlotNumbers
func TestProcessProposerSlashings_UnmatchedShardIds
func TestProcessProposerSlashings_UnmatchedBlockRoots
```

Then for your happy path test would just be `TestProcessProposerSlashings_AppliesCorrectStatus`

Reply:
+1